pl_bolts/datamodules/binary_emnist_datamodule.py

```diff
@@ -0,0 +1,180 @@
+                                          (as in the EMNIST documentation) of
+                                          validation dataset is disbursed based on the ``split``.
+            num_workers (int): How many workers to use for loading data
+            normalize (bool): If ``True``, applies image normalize.
+            batch_size (int): How many samples per batch to load.
+            seed (int): Random seed to be used for train/val/test splits.
+            shuffle (bool): If ``True``, shuffles the train data every epoch.
+            pin_memory (bool): If ``True``, the data loader will copy Tensors into
+                               CUDA pinned memory before returning them.
+            drop_last (bool): If ``True``, drops the last incomplete batch.
+        """
+        if not _TORCHVISION_AVAILABLE:  # pragma: no cover
+            raise ModuleNotFoundError(
+                'You want to use EMNIST dataset loaded from `torchvision` which is not installed yet.'
+            )
+
+        super(BinaryEMNISTDataModule, self).__init__(  # type: ignore[misc]
+            data_dir=data_dir,
+            val_split=val_split,
+            num_workers=num_workers,
+            normalize=normalize,
+            batch_size=batch_size,
+            seed=seed,
+            shuffle=shuffle,
+            pin_memory=pin_memory,
+            drop_last=drop_last,
+            *args,
+            **kwargs,
+        )
+        self.split = split
+        self._check_and_update()
+
+    def _check_and_update(self):
+        if self._emnist_split_exists:
+            self._update_val_split()
+
+    def _update_val_split(self):
+        if (self.val_split is None):
+            if self.split_metadata.get('validation'):
+                self.val_split = int(self.split_metadata.get('num_test'))
+            else:
+                self.val_split = self._DEFAULT_NO_VALIDATION_VAL_SPLIT
+
+    @property
+    def _emnist_split_exists(self) -> bool:
+        """Checks if the split exists in emnist or not."""
+        if not any(s == self.split for s in self.dataset_cls.splits):
+            allowed_values = ', '.join([f'"{v}"' for v in self.dataset_cls.splits])
+            raise ValueError(
+                f'Invalid value provided for split (="{self.split}"). ' + f'Allowed splits are: {allowed_values}'
+            )
+        return True
+
+    @property
+    def split_metadata(self):
+        return self.dataset_cls._metadata.get('splits').get(self.split)
+
+    @property
+    def num_classes(self) -> int:
+        """
+        Return:
+           - for ``byclass``: 62
+           - for ``bymerge``: 47
+           - for ``balanced``: 47
+           - for ``digits``: 10
+           - for ``letters``: 47
+           - for ``mnist``: 10
+        """
+        # The _metadata is only added to EMNIST dataset
+        # to get split-specific metadata.
+        return self.split_metadata.get('num_classes')
```

Comment:
Shall we utilise torchvision's attribute here so that we can get rid of a metadata key `EMNIST_METADATA["splits"][...]["num_classes"]`?https://github.com/pytorch/vision/blob/015378434c1e4be778b2d65eb86f227db20bc8bf/torchvision/datasets/mnist.py#L281
```suggestion
        return len(self.dataset_cls.classes_split_dict[self.split])
```

Reply:
@akihironitta The EMNIST dataset class (from torchvision) does not provide an easy way to access all the content as in Table-II of the paper. I believe, it is very helpful in making it available as a `dict` to the user. This let's you quickly check different aspects of the dataset. More importantly, EMNIST_METADATA is used to run some tests and determine default-validation dataset size for a split.

About using the `classes_split_dict`, yes that can be done (as you showed) and perhaps making `num_classes` dependent only on the core `EMNIST` attributes is a better idea, as this is more organic.

```python
from torchvision.datasets import EMNIST
dict((k, len(v)) for k, v in EMNIST.classes_split_dict.items())
# {'balanced': 47,
#  'byclass': 62,
#  'bymerge': 47,
#  'digits': 10,
#  'letters': 27,
#  'mnist': 10}
```

:warning: However, I would still advise to retain the EMNIST_METADATA as it is, since, the length of the default validation data is determined based on the test data and the split. The as-is version of EMNIST does not provide this information (AFAIK). I added the EMNIST_METADATA as an easy means to access Table-II as well.